spec/models/spree/variant_spec.rb

```diff
@@ -525,6 +525,26 @@ module Spree
           expect(product.master).to be_valid
         end
       end
+
+      context "when the product's unit is non-weight" do
+        before do
+          product.update_attribute :variant_unit, 'volume'
+          product.reload
+          variant.reload
+        end
+
+        it "sets weight to decimal before save if it's integer" do
+          variant.weight = 1
+          variant.save!
+          expect(variant.weight).to eq 1.0
+        end
+
+        it "sets weight to 0.0 before save if it's nil" do
+          variant.weight = nil
+          variant.save!
+          expect(variant.weight).to eq 0.0
+        end
```

Comment:
Cool :ok_hand:. We could test a couple of other cases here where the input is a string, as these unit tests are quite cheap.

What happens when:
```ruby
variant.weight = "123"
```
or...
```ruby
variant.weight = "BANANAS!"
```

Expect the unexpected :sweat_smile:

Reply:
Will add that test case in!